commit 20a435df7093fb0048bf6471e9ca6f3fc17ee3b6
Author: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date:   Wed Dec 18 02:21:50 2013 -0500

    uscan checks for likely upstream signatures if none are known
    
    uscan tries to fetch the usual suffixes (.asc, .gpg, .pgp, .sig)
    appended to the tarball URL to see if we can find a likely-looking
    cryptographic signature.
    
    If one is found, we suggest that the package maintainer to investigate
    it and encourage them set up future checks.

diff --git a/scripts/uscan.pl b/scripts/uscan.pl
index 0ffe9f2..6cdce00 100755
--- a/scripts/uscan.pl
+++ b/scripts/uscan.pl
@@ -1412,6 +1412,16 @@ EOF
 	       '--keyring', 'debian/upstream-signing-key.pgp',
 	       "$destdir/$newfile_base.pgp", "$destdir/$newfile_base") >> 8 == 0
 		 or uscan_die("$progname warning: OpenPGP signature did not verify.\n");
+    } else {
+	print "-- Checking for common possible upsteam OpenPGP signatures\n" if $verbose;
+	foreach my $suffix (qw(asc gpg pgp sig)) {
+	    my $sigrequest = HTTP::Request->new('GET' => "$upstream_url.$suffix");
+	    my $sigresponse = $user_agent->request($sigrequest);
+	    if ($sigresponse->is_success()) {
+		uscan_warn "$pkg: Possible OpenPGP signature found at:\n   $upstream_url.$suffix.\n  Please consider adding opts=pgpsigurlmangle=s/\$/.$suffix/\n  to debian/watch.  see uscan(1) for more details.\n";
+		last;
+	    }
+	}
     }
 
     if ($repack and $newfile_base =~ /^(.*)\.(tar\.bz2|tbz2?)$/) {
